server/core/client-manager.js

```diff
@@ -0,0 +1,32 @@
+const EventEmitter = require('events');
+const Client = require('./client');
+
+module.exports = class ClientManager extends EventEmitter {
```

Comment:
Extending EventEmitter is interesting approach.
We can see that clientManager contains in itself socketIo which is another EventEmitter by itself. So we have kind of a funky situation of event-driven inception 😆 . When a new client is added socketIo will trigger clientManager, which will pass back execution context to socketIo. 

Anyway, thumbs up here 👍